endpoint.go

```diff
@@ -690,7 +690,7 @@ func (ep *endpoint) DataScope() string {
 	return ep.getNetwork().DataScope()
 }
 
-func (ep *endpoint) assignAddress() error {
+func (ep *endpoint) assignAddress(adOptions map[string]string) error {
```

Comment:
I'd prefer to have an `EndpointCreateOptionIpam()` setter  for `CreateEndpioint()` logic which would set a preferred address and a map of options for this endpoint. Similarly to `NetworkOptionIpam()` for network create.

The this function could directly access it as `ipamAllocateVersion()` does for network.